test/unit/api_spec.js

```diff
@@ -822,6 +823,63 @@ describe('api', function() {
         done.fail(reason);
       });
     });
+
+    it('gets non-existent permissions', function(done) {
+      doc.getPermissions().then(function(permissions) {
+        expect(permissions).toEqual(null);
+
+        done();
+      }).catch(function(reason) {
```

Comment:
Here, and elsewhere below, the slightly less verbose `}).catch(done.fail);` ought to work.